beacon_node/lighthouse_network/src/listen_addr.rs

```diff
@@ -6,17 +6,26 @@ use serde::{Deserialize, Serialize};
 /// A listening address composed by an Ip, an UDP port and a TCP port.
 #[derive(Clone, Debug, Serialize, Deserialize)]
 pub struct ListenAddr<Ip> {
+    /// The IP address we will listen on.
     pub addr: Ip,
-    pub udp_port: u16,
+    /// The UDP port that discovery will listen on.
+    pub disc_port: u16,
+    /// The UDP port that QUIC will listen on.
+    pub quic_port: u16,
+    /// The TCP port that libp2p will listen on.
     pub tcp_port: u16,
 }
 
 impl<Ip: Into<IpAddr> + Clone> ListenAddr<Ip> {
-    pub fn udp_socket_addr(&self) -> SocketAddr {
-        (self.addr.clone().into(), self.udp_port).into()
+    pub fn discovery_socket_addr(&self) -> SocketAddr {
+        (self.addr.clone().into(), self.disc_port).into()
     }
 
-    pub fn tcp_socket_addr(&self) -> SocketAddr {
+    pub fn quic_socket_addr(&self) -> SocketAddr {
+        (self.addr.clone().into(), self.quic_port).into()
+    }
+
+    pub fn libp2p_socket_addr(&self) -> SocketAddr {
```

Comment:
```suggestion
    pub fn tcp_socket_addr(&self) -> SocketAddr {
```
so I realized I suggested `libp2p` over `listening` but this change doesn't seem correct to me either. 

`quic` and `tcp` are both transports over which `libp2p` can work. A `libp2p` socket address could be a `quic` one, whereas `tcp` and `quic` unequivocally refer to what we are trying to convey